scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+        output_file(str): output JSONL file
+    """
+    json_files = [f for f in os.listdir(json_directory) if f.endswith('.jsonl')]
+    with open(output_file, 'w') as outfile:
+        for file_name in json_files:
+            with open(os.path.join(json_directory, file_name), 'r') as infile:
+                for line in infile:
+                    outfile.write(line)
+    print('JSON files have been combined into a JSONL file.')
+
+
+def run_query(
+    q: str,
+    method: str,
+    cursor: Optional[Cursor] = None,
+    spark: Optional[SparkSession] = None,
+    collect: bool = True
+) -> Optional[Union[List[Row], DataFrame, SparkDataFrame]]:
+    """Run SQL query via databricks-connect or databricks-sql.
+
+    Args:
+        q (str): sql query
+        method (str): select from dbsql and dbconnect
+        cursor (Cursor): connection.cursor
+        spark (SparkSession): spark session
+        collect (bool): whether to get the underlying data from spark dataframe
+    """
+    if method not in ['dbsql', 'dbconnect']:
+        raise ValueError(f'Unrecognized method: {method}')
+
+    if method == 'dbsql':
+        if cursor is None:
+            raise ValueError(f'cursor cannot be None if using method dbsql')
+        cursor.execute(q)
+        if collect:
+            return cursor.fetchall()
+
+    elif method == 'dbconnect':
+        if spark == None:
+            raise ValueError(f'sparkSession is required for dbconnect')
+        df = spark.sql(q)
+        if collect:
+            return df.collect()
+        return df
+
+    return None
+
+
+def get_args(signed: List, json_output_path: str, columns: List,
+             cf_collect_type: str) -> Iterable:
+    for i, r in enumerate(signed):
+        yield (i, r.url, json_output_path, columns, cf_collect_type)
+
+
+def download(ipart: int,
+             url: str,
+             json_output_path: str,
+             columns: Optional[List] = None,
+             resp_format: str = 'arrow',
+             compressed: bool = False) -> None:
+    """Thread download presigned url and save to jsonl locally.
+
+    Args:
+        ipart (int): presigned url id
+        url (str): presigned url
+        json_output_path (str): directory to save the ipart_th segment of dataframe
+        columns (list): schema to save to json
+        resp_format (str): whether to use arrow or json when collect
+        compressed (bool): if data is compressed before downloading. Need decompress if compressed=True.
+    """
+    resp = requests.get(url)
+    if resp.status_code == 200:
+        if resp_format == 'json':
+            data = resp.json()
+            pd.DataFrame(data, columns=columns).to_json(os.path.join(
+                json_output_path, 'part_' + str(ipart) + '.jsonl'),
+                                                        orient='records',
+                                                        lines=True)
+            return
+
+        if compressed:
```

Comment:
Can you please add a comment that this will get called when `resp_format` is `arrow`? Or did I misunderstood the code!

Reply:
comment added.